src/backend/distributed/shared_library_init.c

```diff
@@ -194,6 +195,16 @@ CreateRequiredDirectories(void)
 static void
 RegisterCitusConfigVariables(void)
 {
+	DefineCustomIntVariable(
+		"citus.node_connection_timeout",
+		gettext_noop("Sets the maximum duration to connect to worker nodes."),
+		NULL,
+		&NodeConnectionTimeout,
+		5000, 1000, INT_MAX,
```

Comment:
Such max values are rather likely to overflow. I'd rather set a maximum of an hour or so.   Also, shouldn't this be GUC_UNIT_MS?